sdk/hash/src/lib.rs

```diff
@@ -64,15 +68,25 @@ impl AsRef<[u8]> for Hash {
     }
 }
 
+fn to_base58(h: &Hash, f: &mut fmt::Formatter) -> fmt::Result {
```

Comment:
nit: can we call this something with `write`, ie `write_as_base58` and move the parameters to look more like https://doc.rust-lang.org/std/fmt/fn.write.html? ie
```suggestion
fn write_as_base58(f: &mut fmt::Formatter, h: &Hash) -> fmt::Result {
```